Add description of Azure spot VMs for kubectl-gs template nodepool (#1039)
giantswarm/docs: src/content/ui-api/kubectl-gs/template-nodepool.md

### Azure specific

- `--azure-vm-size` - Azure VM size to use for workers (e.g. *Standard_D4s_v3*).

> - `--azure-spot-vms` - Wheter to use Spot VMs for this Node Pool or not (defaults to false that means not to use Spot VMs).

Reviewer: ```suggestion
- `--azure-spot-vms` - Whether to use spot VMs for this node pool (defaults to false which means not to use spot VMs).
```

> - `azure-spot-max-price` - Max hourly price in USD to pay for one spot VM. If the current price for the VM is exceeded, the VM is deleted.

Reviewer 2: ```suggestion
- `--azure-spot-max-price` - Max hourly price in USD to pay for one spot VM. If the current price for the VM is exceeded, the VM is deleted.
```

> - `azure-spot-max-price` - Max hourly price in USD to pay for one Spot VM. If the current price for the VM is exceeded, the VM is deleted.

Reviewer: ```suggestion
- `azure-spot-max-price` - Max hourly price in USD to pay for one spot VM. If the current price for the VM is exceeded, the VM is deleted.
```

Reviewer: I'd like us to document what the default is. I. e. what price is used if this flag is not used.

> - `--azure-spot-max-price` - Max hourly price in USD to pay for one spot VM. If the current price for the VM is exceeded, the VM is deleted. If not set, the on demand price for the same instance size is used as max value.

Reviewer: ```suggestion
- `--azure-spot-max-price` - Max hourly price in USD to pay for one spot VM. If the current price for the VM is exceeded, the VM is deleted. If not set, the on-demand price for the same machine size is used as limit.
```
